modules/fluid_properties/src/materials/BrineFluidPropertiesTestMaterial.C

```diff
@@ -0,0 +1,70 @@
+/****************************************************************/
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*          All contents are licensed under LGPL V2.1           */
+/*             See LICENSE for full restrictions                */
+/****************************************************************/
+
+#include "BrineFluidPropertiesTestMaterial.h"
+
+template<>
+InputParameters validParams<BrineFluidPropertiesTestMaterial>()
+{
+  InputParameters params = validParams<Material>();
+  params.addRequiredCoupledVar("pressure", "pressure (Pa)");
+  params.addRequiredCoupledVar("temperature", "temperature (K)");
+  params.addRequiredCoupledVar("xnacl", "NaCl mass fraction (-)");
+  params.addRequiredParam<UserObjectName>("fp", "The name of the user object for fluid properties");
+  params.addParam<bool>("vapor", false, "Flag to calculate vapor pressure");
+  params.addClassDescription("Material to test brine properties");
+  return params;
+}
+
+BrineFluidPropertiesTestMaterial::BrineFluidPropertiesTestMaterial(const InputParameters & parameters) :
+    Material(parameters),
+    _pressure(coupledValue("pressure")),
+    _temperature(coupledValue("temperature")),
+    _xnacl(coupledValue("xnacl")),
+
+    _rho(declareProperty<Real>("density")),
+    _mu(declareProperty<Real>("viscosity")),
+    _h(declareProperty<Real>("enthalpy")),
+    _cp(declareProperty<Real>("cp")),
+    _e(declareProperty<Real>("e")),
+    _k(declareProperty<Real>("k")),
+    _solubility(declareProperty<Real>("solubility")),
+    _pvap(declareProperty<Real>("vapor")),
+    _rho_halite(declareProperty<Real>("halite_density")),
+    _cp_halite(declareProperty<Real>("halite_cp")),
+    _h_halite(declareProperty<Real>("halite_enthalpy")),
+    _vapor(getParam<bool>("vapor")),
+
+    _fp(getUserObject<BrineFluidProperties>("fp")),
+    _water_fp(_fp.getComponent(0)),
+    _halite_fp(_fp.getComponent(1))
```

Comment:
Looks much better with this - thanks. I have fixed up all of these suggestions. After this gets merged I'll nuke the water and brine materials in `porous_flow`